server/block/campfire.go

```diff
@@ -112,6 +112,14 @@ func (c Campfire) Activate(pos cube.Pos, _ cube.Face, w *world.World, u item.Use
 	if held.Empty() {
 		return false
 	}
+
+	if _, ok := held.Item().(item.Shovel); ok {
```

Comment:
This should take durability from the shovel. You can implement this interface to get that automatically: https://github.com/df-mc/dragonfly/blob/master/server/item/shovel.go#L41

Reply:
Also check first that the campfire isn't already extinguished